### PR TITLE
`CustomerInfoManager`: fixed `Lock` usage

### PR DESCRIPTION
**Pull request by NachoSoto**
I noticed that there was a single `.unlock` call. I decided to refactor it by using the new `Lock`.
This class could use some more love to make sure the whole thing is thread-safe, but for now this should fix issues with the lock getting unlocked.

### CODE REVIEW

File: Purchases/Identity/CustomerInfoManager.swift [hunk 1]
```diff
@@ -168,28 +168,29 @@ class CustomerInfoManager {
...
+
+            if lastSentCustomerInfo != nil {
+                Logger.debug(Strings.customerInfo.sending_updated_customerinfo_to_delegate)
+            } else {
+                Logger.debug(Strings.customerInfo.sending_latest_customerinfo_to_delegate)
+            }
 
-        self.lastSentCustomerInfo = customerInfo
-        operationDispatcher.dispatchOnMainThread {
-            self.customerInfoCacheLock.unlock()
```
Comment (NachoSoto) [inline Purchases/Identity/CustomerInfoManager.swift:191]: Oops.

**Review by NachoSoto — COMMENTED** [review]
I recommend looking at the diff without space changes

File: Purchases/Identity/CustomerInfoManager.swift (hunk 1 above)
Comment (taquitos) [reply Purchases/Identity/CustomerInfoManager.swift:191]: 😵

**Review by taquitos — APPROVED** [review]
🔒 🔓 🔓

**Pull request merged**